### PR TITLE
chore: migration to the new redocly portal

### PR DESCRIPTION
**Pull request by RomanHotsiy**
⚠️ WIP

Migration TODO:

- Move i18n files to correct folder
  - I moved only one folder manually `concepts/introduction`
- Migrate links
  - absolute paths in translation folders should start from locale,
    see content/@i18n/ja/concepts/introduction/crypto-wallets.md file for example (the last line)
- Generate redirects.yaml
- Replace snippets (include with partial)
  - Feature request: configure snippet folder - not started yet by Redocly
- Migrate html pages to react pages, with translations
  - Example usage in index.page.tsx
  - Script for generationg translations.yaml is - in-progress by Redocly
  - Potentially we'll add native support for .po, .mo files
- Interactive pages
  - Event hook for page visited (similar to document contentdomready) - in progress by Redocly
- Autogenerate index pages
  - Created example plugin, uses the order from the sidebars (it uses some internals but we'll expose them later)

### CODE REVIEW

File: content/_code-samples/escrow/websocket/ledger-response-expiration.json [hunk 1]
```diff
@@ -4,13 +4,13 @@
   "type": "response",
   "result": {
     "ledger": {
-      ... (trimmed) ...
+      # ... (trimmed) ...
```
Comment (mDuo13) [inline content/_code-samples/escrow/websocket/ledger-response-expiration.json:7]: What is going on here? Obviously the original isn't proper JSON syntax because it's abbreviated, but `#` isn't valid JSON syntax either.

File: content/privacy-policy.md [hunk 2]
```diff
@@ -117,14 +117,14 @@ If you have any questions about this Privacy Policy, You can contact us:
...
+15, Ringtee
+Kuressaare
 Estonia 93815
 
-**Tallinn Office**  
-802, Lõõtsa tn 5  
-Tallinn  
+**Tallinn Office**
+802, Lõõtsa tn 5
+Tallinn
```
Comment (mDuo13) [inline content/privacy-policy.md:127]: I think your editor was overzealous with removing trailing spaces here. This is actually intended syntax, using a little-known feature of [the original Markdown spec](https://daringfireball.net/projects/markdown/syntax#p) to distinguish intentional line breaks from hard-wrapped text:

> When you do want to insert a `<br />` break tag using Markdown, you end a line with two or more spaces, then type return.

File: content/@i18n/ja/concepts/introduction/index.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+---
+metadata:
+  indexPage: true
+---
+
+# カテゴリ
```
Comment (mDuo13) [inline content/@i18n/ja/concepts/introduction/index.md]: I'm not sure what happened here, but the category name has not been translated correctly. (It says "Category" instead of "Introduction".) Maybe the translation was based on the placeholder, variable name, or something by accident?

```suggestion
# 基本
```

**Comment by mDuo13** [comment]
I [implemented a tool for generating a sidebar in Redocly format](https://github.com/mDuo13/xrpl-dev-portal/commit/65463ac1733afbaf3c3bb7f6207d19da01cfafdb) based on the Dactyl config, but it needs to be run on a branch with a non-broken Dactyl build. 

So the workflow for using it right now is kind of funky and involves 3 branches, but hopefully we won't need to do that much.

Details below just in case someone else wants to generate the sidebar. You probably don't need to do this since I already pushed the better sidebar to this branch.

1. switch to `master` branch
2. check out the one `tool/filter_enforce_filenames.py` file from my dev branch and uncomment the code on line 97 to run the sidebar generator (optionally specify a different top-level page; could do this multiple times to get multiple sidebars for different sections rather than one for the whole site)
3. run `dactyl_build`, save/stash the `exported-sidebars.yml` file somewhere
4. switch back to the `redocly-migration` branch. replace the sidebars file with the updated one

File: content/_code-samples/escrow/websocket/ledger-response-expiration.json (hunk 1 above)
Comment (RomanHotsiy) [reply content/_code-samples/escrow/websocket/ledger-response-expiration.json:7]: This is bug in portal, there is a code that tries to parse every json with yaml parser and crashes for invalid format.
We'll fix it in the next release.

File: content/privacy-policy.md (hunk 2 above)
Comment (RomanHotsiy) [reply content/privacy-policy.md:127]: oh, I see. Another way to do an intentional link break is to use `/` at the end of the line. Not sure if this is described in any spec but it works like this with markdoc.

File: content/@i18n/ja/concepts/introduction/index.md (hunk 3 above)
Comment (RomanHotsiy) [reply content/@i18n/ja/concepts/introduction/index.md]: maybe I copied it from wrong place

**Comment by amarantha-k** [comment]
@RomanHotsiy I've merged updates from `master` to the `redocly-migration` branch. Could you rebase so this PR only includes commits related to Redocly migration?

File: assets/vendor/fontawesome/css/font-awesome.min.css
Comment (mDuo13) [inline assets/vendor/fontawesome/css/font-awesome.min.css:1]: Why is this deleted?

**Comment by mDuo13** [comment]
Closing this version of the PR since we don't plan to merge this as-is. (We still plan to move to Redocly and I have cherry-picked some of the commits from here into the `redocly-migration` branch.)
